### ISSUE TITLE
Document the public API

### ISSUE THREAD

**Issue by Asuza** [JSJitsu/ai-battle-website#99]
The API, while undocumented, is currently public. It would be nice to have a page dedicated to explaining the various routes and payloads for developers to use if they so choose.

**Comment by scampiuk** [comment]
I would like to be assigned this issue please :heart:

**Comment by scampiuk** [comment]
Just an update, this is coming along now
[-] documented most of the endpoints
[x] added route to view the documenation on the website
[  ] need to add tests

what I'm going to do I think is open a PR early on this, there are things I don't know and I will obviously have missed because, well, it's not my app :D 

However, coming together.

Regarding the testing, I'm wondering if it's ok if I open in a different PR?  Just stops 'one big pr' hitting

Cheers,

( sneak peak of the documentation viewer being served from inside the docker.. )

![image](https://user-images.githubusercontent.com/1496616/95665254-8ad89c80-0b46-11eb-96a5-f6ce1d256253.png)

**Comment by Asuza** [comment]
Absolutely. The primary goal of this issue was to document, not test. It's looking good!

**Comment by Asuza** [comment]
@scampiuk, just wanted to check in since October is almost over. Can you make a PR for this?

**Comment by scampiuk** [comment]
Yes sorry, I hate it when I sign up for things then forget I've got a job and two kids and stuff! :D I've got this week off work so i'll push the document up over the next 48 hrs
